src/preferences/dialog/dlgprefcontrols.cpp

```diff
@@ -681,6 +701,30 @@ void DlgPrefControls::slotApply() {
 
     int configSPAutoReset = BaseTrackPlayer::RESET_NONE;
 
+    // screensaver mode update
+    int inhibitcombo = comboBoxScreensaver->itemData(
+            comboBoxScreensaver->currentIndex()).toInt();
+    int inhibitsettings = m_pConfig->getValue<int>(ConfigKey("[Config]","InhibitScreensaver"));
+    if (inhibitcombo != inhibitsettings) {
+        m_pConfig->set(ConfigKey("[Config]", "InhibitScreensaver"), ConfigValue(inhibitcombo));
+
+        mixxx::ScreenSaverPreference inhiOld = mixxx::ScreenSaverPreference(inhibitsettings);
+        if (inhiOld == mixxx::ScreenSaverPreference::PREVENT_ON) {
+            mixxx::ScreenSaverHelper::uninhibit();
+        } else if (inhiOld == mixxx::ScreenSaverPreference::PREVENT_ON_PLAY) {
+            mixxx::ScreenSaverHelper::inhibitOnCondition(false);
```

Comment:
can't we call a plain uninhibit in this case as well? 
Maybe we need to move the check if it is already uninhibited inside the mixxx::ScreenSaverHelper::uninhibit(); 
and simplify this code here much.